cleverpush/src/main/java/com/cleverpush/banner/AppBannerCarouselAdapter.java

```diff
@@ -476,6 +477,23 @@ private void composeHtmlBLock(LinearLayout body, BannerHTMLBlock block) {
     webView.getSettings().setJavaScriptEnabled(true);
     webView.setVerticalScrollBarEnabled(false);
     webView.setHorizontalScrollBarEnabled(false);
+
+    webView.setWebViewClient(new WebViewClient() {
+      @Override
+      public void onPageFinished(WebView view, String url) {
+        super.onPageFinished(view, url);
+        webView.evaluateJavascript(
```

Comment:
whats this about?

Reply:
Setting the WebViewClient to inject JavaScript after the page is loaded. 

because for `trackEvent` we are using `trackEventStringified` and for `trackClick` we are using `trackClickStringified`

You can see for how we use in Edit Via HTML: https://github.com/cleverpush/cleverpush-android-sdk/blob/39e55a6110990dee94409b55ae3425104bd68a7b/cleverpush/src/main/java/com/cleverpush/banner/AppBannerCarouselAdapter.java#L518

For Edit Via HTML we use html content. For HTML block we use url. 

composeHtmlBanner works fine because we explicitly define the CleverPush.trackEvent and CleverPush.trackClick methods in the HTML content

in composeHtmlBlock we are loading the content from a URL, the JavaScript functions is not available in that content by default. So we have to inject the necessary JavaScript functions into the WebView after the URL content is loaded.